### PR TITLE
Site Profiler: Open get report form when clicking the Metrics CTA

### PR DESCRIPTION
**Pull request by epeicher**
Fixes https://github.com/Automattic/dotcom-forge/issues/6909

## Proposed Changes

* Adds functionality to open the Get Report form when clicking the menu CTA
* Adds functionality to close the form when clicking on the `X` icon
* The arrow icon in the top-right of the form has been replaced by a `X` icon to indicate closing
* The form is fixed positioned at the bottom of the viewport

CleanShot 2024-05-10 at 18 27 17@2x

## Testing Instructions

* Apply this patch to your environment
* Navigate to the site-profiler and analyze a website
* Click on the main CTA button in the basic metrics menu 
CleanShot 2024-05-10 at 18 23 36@2x
* Check the form is displayed at the bottom and that you can scroll through the results
* Click on the `X` icon in the form
* Check the form is closed as expected
* Check the style and behavior is similar to the production version

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
~~- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?~~
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
~~- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?~~
~~- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?~~

### CODE REVIEW

File: client/site-profiler/components/get-report-form/index.tsx [hunk 1]
```diff
@@ -98,89 +98,89 @@ export function GetReportForm( {
...
+									name="termsAccepted"
+									className="terms-checkbox"
+									checked={ isTermsChecked }
+									onChange={ handleTermsChange }
+									label={ translate(
+										`By submitting your details, you agree to WordPress.com‘s Privacy Policy and Terms of Service. You also consent to receiving occasional updates and offers. You can unsubscribe from these communications at any time through the instructions.`
+									) }
+								/>
+								{ errors?.termsAccepted && (
+									<FormInputValidation isError={ !! errors } text={ errors.termsAccepted } />
```
Comment (WBerredo) [inline client/site-profiler/components/get-report-form/index.tsx]: Nit: Is this check `isError={ !! errors }` needed? Considering its checked on the previous line.
```suggestion
									<FormInputValidation isError text={ errors.termsAccepted } />
```

The same for the previous components.

**Review by WBerredo — COMMENTED** [review]
CleanShot 2024-05-10 at 16 36 42@2x

I think this button was different to give the impression of sliding up and down. I think the current show/hide behavior is a bit abrupt so I would also prefer the sliding transition.

File: client/site-profiler/components/get-report-form/index.tsx (hunk 1 above)
Comment (epeicher) [reply client/site-profiler/components/get-report-form/index.tsx]: Not needed here, removing it

**Comment by epeicher** [comment]
> I think this button was different to give the impression of sliding up and down. I think the current show/hide behavior is a bit abrupt so I would also prefer the sliding transition.

I agree that this form will benefit from a sliding animation, and the behavior is currently abrupt. I will work on that

File: client/site-profiler/components/get-report-form/index.tsx (hunk 1 above)
Comment (epeicher) [reply client/site-profiler/components/get-report-form/index.tsx]: 3a474a8716

**Comment by epeicher** [comment]
I have implemented slide-up and slide-down animations, but I have observed that the behavior is not consistent among browsers:
- It works on Chrome
- It works for slide-up but not for slide-down in Firefox and Safari
- It does not seem to work at all in Arc

This is the Chrome behavior:
CleanShot 2024-05-13 at 12 39 55

File: client/site-profiler/components/get-report-form/styles.scss [hunk 2]
```diff
@@ -1,13 +1,47 @@
...
+	}
+}
+@keyframes slideDown {
+	from {
+		display: block;
+		transform: translateY(0);
+	}
+	to {
+		display: none;
+		transform: translateY(100%);
```
Comment (cpapazoglou) [inline client/site-profiler/components/get-report-form/styles.scss:33]:
```suggestion
	&--hidden {
		animation: 500ms ease-out slideDown;
		transform: translateY(100%);
	}
}

@keyframes slideUp {
	from {
		transform: translateY(100%);
	}
	to {
		transform: translateY(0);
	}
}
@keyframes slideDown {
	from {
		transform: translateY(0);
	}
	to {
		transform: translateY(100%);
```

**Comment by epeicher** [comment]
A follow-up task has been created to investigate a fix so the animation is consistent between browsers:
 - https://github.com/Automattic/dotcom-forge/issues/7130

File: client/site-profiler/components/get-report-form/styles.scss (hunk 2 above)
Comment (epeicher) [reply client/site-profiler/components/get-report-form/styles.scss:33]: I have tried this, but it displays the form initially U02M8SCS6RE/F073J5UBW8H-slack-file

We will investigate this and other options as part of the linked task https://github.com/Automattic/dotcom-forge/issues/7130

**Review by gcsecsey — APPROVED** [review]
LGTM and works great!

**Pull request merged**